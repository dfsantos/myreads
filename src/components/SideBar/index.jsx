import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import AppBar from 'material-ui/AppBar';
import Drawer from 'material-ui/Drawer';
import MenuItem from 'material-ui/MenuItem';

const propTypes = {
  open: PropTypes.bool,
  onClose: PropTypes.func.isRequired,
};

const defaultProps = {
  open: false,
};

function SideBar({ open, onClose }) {
  return (
    <Drawer docked={false} open={open} onRequestChange={onClose}>
      <AppBar title="BookFlix" onLeftIconButtonTouchTap={onClose} />
      <Link className="menu-item-link" to="/">
        <MenuItem>My Shelves</MenuItem>
      </Link>
      <Link className="menu-item-link" to="/search">
        <MenuItem>Search</MenuItem>
      </Link>
    </Drawer>
  );
}

SideBar.propTypes = propTypes;
SideBar.defaultProps = defaultProps;

export default SideBar;
