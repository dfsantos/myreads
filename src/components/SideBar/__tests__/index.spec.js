import React from 'react';

import SideBar from '../index';

const onCloseListener = jest.fn();

it('should render without errors', () => {
  shallow(<SideBar open onClose={onCloseListener} />);
});

it('snapshot of SideBar closed', () => {
  const wrapper = shallow(<SideBar onClose={onCloseListener} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('snapshot of SideBar opened', () => {
  const wrapper = shallow(<SideBar open onClose={onCloseListener} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
