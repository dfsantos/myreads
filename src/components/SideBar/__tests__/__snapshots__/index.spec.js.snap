// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of SideBar closed 1`] = `
<Drawer
  disableSwipeToOpen={false}
  docked={false}
  onRequestChange={[Function]}
  open={false}
  openSecondary={false}
  swipeAreaWidth={30}
  width={null}
  zDepth={2}
>
  <AppBar
    onLeftIconButtonTouchTap={[Function]}
    showMenuIconButton={true}
    title="BookFlix"
    zDepth={1}
  />
  <Link
    className="menu-item-link"
    replace={false}
    to="/"
  >
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    >
      My Shelves
    </MenuItem>
  </Link>
  <Link
    className="menu-item-link"
    replace={false}
    to="/search"
  >
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    >
      Search
    </MenuItem>
  </Link>
</Drawer>
`;

exports[`snapshot of SideBar opened 1`] = `
<Drawer
  disableSwipeToOpen={false}
  docked={false}
  onRequestChange={[Function]}
  open={true}
  openSecondary={false}
  swipeAreaWidth={30}
  width={null}
  zDepth={2}
>
  <AppBar
    onLeftIconButtonTouchTap={[Function]}
    showMenuIconButton={true}
    title="BookFlix"
    zDepth={1}
  />
  <Link
    className="menu-item-link"
    replace={false}
    to="/"
  >
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    >
      My Shelves
    </MenuItem>
  </Link>
  <Link
    className="menu-item-link"
    replace={false}
    to="/search"
  >
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    >
      Search
    </MenuItem>
  </Link>
</Drawer>
`;
